### ISSUE TITLE
Default media players start with their volume slider set to local

### ISSUE THREAD

**Issue by JackTheFoxOtter** [Yellow-Dog-Man/Resonite-Issues#2006]
### Describe the Underlying Issue

This is a regression from a couple weeks ago. The media players used to spawn with the volume slider set to global, however, this has changed, now they are local.

### Expected Behavior

The default audio slider state should be global.

### Screenshots / Video

Current video player interface immediately after spawning:
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/35171546/df92d52b-c3fb-45c4-8f23-bad4acd26b85)

### Additional Context

This is problematic because this makes it impossible to mute the media for everyone by default. Prior to this, whenever I spawned out a video in a large world, I'd first set the video player's global audio to 0, and then instruct others that want to watch to switch it to local and pick a comfortable volume. This is now impossible. Switching audio modes is a local action, so even if I switch to global first and turn down the volume, for everyone else it will still be set to local at 0.5 volume, unless they also switch the volume to global manually.

**Comment by Zyzyl** [comment]
This was a deliberate change Ryuvi reported in the Devlog channel https://discord.com/channels/1040316820650991766/1154514014563483759/1233241897258979459

**Comment by JackTheFoxOtter** [comment]
Ah. Well this is not a good change in my eyes, as this makes it impossible to mute the video player for everyone currently in the world. This currently prevents me from spawning out any videos, as I don't want to blast anyone with audio they don't want to hear.

**Comment by JackTheFoxOtter** [comment]
This would be fine for me if the default local volume was set to 0, so people have to "opt in" to listen. But I imagine that's not beneficial for other situations, like lectures where the lecturer spawns out a video and wants everyone to hear it by default.

**Comment by Readun** [comment]
Check what the default global slider state is, as far as I remembered, it was under 25% ?

**Comment by JackTheFoxOtter** [comment]
That does not matter, as the default audio mode is set to local.

**Comment by 5H4D0W-X** [comment]
> This currently prevents me from spawning out any videos

Keep in mind that you can just revert this change and favorite it as a custom video player.

**Comment by BlueCyro** [comment]
> > This currently prevents me from spawning out any videos
> 
> Keep in mind that you can just revert this change and favorite it as a custom video player.

This does present an issue for everyone else who hasn't favorited a new player though. If they spawn out a normal video player, it affects everyone as they can't turn the volume down for everyone else. The volume being at zero on spawn is also rather inconvenient as well.

**Comment by JackTheFoxOtter** [comment]
> Keep in mind that you can just revert this change and favorite it as a custom video player.

Yes, and I will when I get around to it. But I believe this isn't good default behavior and would like to at least mention it.

**Comment by Electronus** [comment]
Why not just have two sliders?

**Comment by BlueCyro** [comment]
> Why not just have two sliders?

Two sliders diverges from the norm that every single other program on your computer uses.

**Comment by JackTheFoxOtter** [comment]
> Why not just have two sliders?

I don't see what that would fix. The default audio mode is **local**. Unless there is a way to change the default audio mode for **other people**, no amount of extra sliders will help, as it woudnd't affect anyone without them actively switching to global audio mode. That's the issue, I need the way to set a video to be muted for everyone unless they opt in.

**Comment by Readun** [comment]
This is a tricky request. 
Some want it to default spawn as global, others as local.
I feel like its better to have local as new users might not know that they are changing the volume for everyone. 
Usually adjusting audio is locally.

The other way would be to favorite and I guess you could also go into the players config to set the local volume higher or lower by default for everyone, if you want.

**Comment by BlueCyro** [comment]
I talked to Ryuvi about this the other night when I had brought this issue up in person. A suggestion we discussed was that the video players could listen for a cloud variable that the user can set (via setting, facet or otherwise) that would dictate the default state of the volume, and the default volume level when another user spawns one.

This proposed setting kind of bridges the gap so that players can opt in or out of respecting these settings. For those who want to respect the audio level that other people spawn their players at, you can leave this at "Global for me", and video players will subsequently spawn with global audio for you. "Local for me" would have everyone's players spawn as local for you. Similar options for volume level would ensue as well. This means that no one way of listening to video players is enforced on everyone.

**Comment by TisFoolish** [comment]
What about just switching it back to global manually then muting that? If people haven't set a local volume level that should still mute them if I'm remembering how the video player is set up (I think it was that global was just the default value of a value user override)

**Comment by BlueCyro** [comment]
> What about just switching it back to global manually then muting that? If people haven't set a local volume level that should still mute them if I'm remembering how the video player is set up (I think it was that global was just the default value of a value user override)

The locality state of the slider is itself local to each user.

**Comment by BlueCyro** [comment]
> > What about just switching it back to global manually then muting that? If people haven't set a local volume level that should still mute them if I'm remembering how the video player is set up (I think it was that global was just the default value of a value user override)
> 
> The locality state of the slider is itself local to each user.

To elaborate, this is so that you can 'subscribe' to what the global setting is, or override it for just yourself.

**Comment by Frooxius** [comment]
There's a thing I proposed when this video player was being originally developed, but that didn't make it in, but that might resolve the issue.

One way to solve this problem is to have the local volume be a percentage of the global one. With a button/element, anyone (including the host) could open the global slider and then adjust everything globally, muting the video or lowering the volume for everyone present when needed. Within that volume each user can then adjust it on their own.

Would this kind of solution help resolve all the problems here?

Another alternative is to simply have a global "mute" option in addition.

**Comment by BlueCyro** [comment]
> There's a thing I proposed when this video player was being originally developed, but that didn't make it in, but that might resolve the issue.
> 
> One way to solve this problem is to have the local volume be a percentage of the global one. With a button/element, anyone (including the host) could open the global slider and then adjust everything globally, muting the video or lowering the volume for everyone present when needed. Within that volume each user can then adjust it on their own.
> 
> Would this kind of solution help resolve all the problems here?
> 
> Another alternative is to simply have a global "mute" option in addition.

I think the cloud variable to let the individual pick what volume state they want to subscribe to by default I think would probably be the more desirable option here. There are cases where the volume might be too quiet, and only allowing the local volume to be a percentage of the global one I can see being an issue for people with quieter audio setups, or who have hearing impairments.

**Comment by TisFoolish** [comment]
> The locality state of the slider is itself local to each user.


Huh, that's odd that the state of the local slider would affect it so when the users don't have any written local data.

**Comment by BlueCyro** [comment]
> > The locality state of the slider is itself local to each user.
> 
> Huh, that's odd that the state of the local slider would affect it so when the users don't have any written local data.

Think of it as a subscription. There is the global volume, and there is your personal "local" volume. Whether you are listening to the global or local volume is a choice you make for yourself and yourself only.

**Comment by JackTheFoxOtter** [comment]
> I feel like its better to have local as new users might not know that they are changing the volume for everyone.

I disagree. It's very easy to explain to a new user how the audio slider works, I've done that countless of times. But as this currently stands, this makes it impossible to start a video as muted for everyone that hasn't opted in to listening. In a world with 20 people all over the place, when I spawn a video to show a friend, I get 18 people mad at me now since I can't turn off the volume for them anymore.

> This proposed setting kind of bridges the gap so that players can opt in or out of respecting these settings. For those who want to respect the audio level that other people spawn their players at, you can leave this at "Global for me", and video players will subsequently spawn with global audio for you. "Local for me" would have everyone's players spawn as local for you. Similar options for volume level would ensue as well. This means that no one way of listening to video players is enforced on everyone.

I'm not sure that would be a good solution. This is kind of a case-by-case situation. Generally the person spawning the audio should be the one to set the default audio for everyone, unless those people then locally override it. Sometimes that might be muting it, to just show one person a funny video without annoying everyone else, sometimes that would be keeping it loud so that everyone in the room can hear it, if you're in a sort of "presenter" role in the current world. If users set the volume slider to always local, cloud variable or not, I cannot mute it for them.

> One way to solve this problem is to have the local volume be a percentage of the global one. With a button/element, anyone (including the host) could open the global slider and then adjust everything globally, muting the video or lowering the volume for everyone present when needed. Within that volume each user can then adjust it on their own.

If the local slider is a multiplier, and the global audio is set to 0, then you wouldn't be able to actually raise your local volume though.

> Another alternative is to simply have a global "mute" option in addition.

That could work. Perhaps a distinct button when you're in global mode to "set for everyone". So I could specify what default local volume I'd like to set for others.

**Comment by BlueCyro** [comment]
> > I feel like its better to have local as new users might not know that they are changing the volume for everyone.
> 
> I disagree. It's very easy to explain to a new user how the audio slider works, I've done that countless of times. But as this currently stands, this makes it impossible to start a video as muted for everyone that hasn't opted in to listening. In a world with 20 people all over the place, when I spawn a video to show a friend, I get 18 people mad at me now since I can't turn off the volume for them anymore.
> 
> > This proposed setting kind of bridges the gap so that players can opt in or out of respecting these settings. For those who want to respect the audio level that other people spawn their players at, you can leave this at "Global for me", and video players will subsequently spawn with global audio for you. "Local for me" would have everyone's players spawn as local for you. Similar options for volume level would ensue as well. This means that no one way of listening to video players is enforced on everyone.
> 
> I'm not sure that would be a good solution. This is kind of a case-by-case situation. Generally the person spawning the audio should be the one to set the default audio for everyone, unless those people then locally override it. Sometimes that might be muting it, to just show one person a funny video without annoying everyone else, sometimes that would be keeping it loud so that everyone in the room can hear it, if you're in a sort of "presenter" role in the current world. If users set the volume slider to always local, cloud variable or not, I cannot mute it for them.
> 
> > One way to solve this problem is to have the local volume be a percentage of the global one. With a button/element, anyone (including the host) could open the global slider and then adjust everything globally, muting the video or lowering the volume for everyone present when needed. Within that volume each user can then adjust it on their own.
> 
> If the local slider is a multiplier, and the global audio is set to 0, then you wouldn't be able to actually raise your local volume though.
> 
> > Another alternative is to simply have a global "mute" option in addition.
> 
> That could work. Perhaps a distinct button when you're in global mode to "set for everyone". So I could specify what default local volume I'd like to set for others.

I think the cloud setting would adequately bridge the gap. Right now we have two reasons for why people want either option.

- Local by default: "I don't want my ears blown out by other peoples' video players"

- Global by default: "I want to be able to turn my video players down for other people"

These are two extremely valid reasons for wanting either option. I often find myself flip flopping between these two states regularly. Sometimes I want to subscribe to what people are spawning in the world, and sometimes I just don't wanna hear the next annoying meme someone spawns out that's loud as heck. This gives people the option and reduces the chances you'll get your ears blown out, or worry about annoying other people with your own player since you can't turn it down for other people.

**Comment by TisFoolish** [comment]
Honestly i feel like setting the global should just also set the default value for unset local volume as well. You might get cases where there are people who are asshats and mess with the global just to mess with those that have yet to set a custom local, but that would be rare. Certainly far rarer than the need to mute the audio for everyone by default

**Comment by JackTheFoxOtter** [comment]
Or perhaps another option would be the following:

Default audio slider mode is local (purple), but the global volume isn't overridden until you touch it to make a change. Before it's touched, it's synced with the global volume. So if I want to set a default, I can switch to gloabal and just change the volume. Everyone else would see their (purple) volume slider move, until they touch it to change it, then they override it. This would actually fix a couple of problem:

1. It would allow someone to deliberately change the default for everyone when spawning a new player
2. Users who don't know how the slider works can't accidentally raise the volume for everyone, as by default they only set an override for themselves.
3. Once you have an override set, it cannot be changed by others anymore, so a muted video can't be unmuted, and a unmuted video can't be muted by others.

**Comment by BlueCyro** [comment]
> Or perhaps another option would be the following:
> 
> Default audio slider mode is local (purple), but the global volume isn't overridden until you touch it to make a change. Before it's touched, it's synced with the global volume. So if I want to set a default, I can switch to gloabal and just change the volume. Everyone else would see their (purple) volume slider move, until they touch it to change it, then they override it. This would actually fix a couple of problem:
> 
>     1. It would allow someone to deliberately change the default for everyone when spawning a new player
> 
>     2. Users who don't know how the slider works can't accidentally raise the volume for everyone, as by default they only set an override for themselves.
> 
>     3. Once you have an override set, it cannot be changed by others anymore, so a muted video can't be unmuted, and a unmuted video can't be muted by others.

This unfortunately doesn't solve the problem global video players already have. It's essentially one button press less for adjusting the local volume as it is now.

**Comment by JackTheFoxOtter** [comment]
> This unfortunately doesn't solve the problem global video players already have. It's essentially one button press less for adjusting the local volume as it is now.

Which problem is that in particular? It sounded like the issue is users accidentally raising the volume for everyone. This would prevent that from happening, as they'd have to deliberately first switch to global mode to change the default.

**Comment by BlueCyro** [comment]
> > This unfortunately doesn't solve the problem global video players already have. It's essentially one button press less for adjusting the local volume as it is now.
> 
> Which problem is that in particular? It sounded like the issue is users accidentally raising the volume for everyone. This would prevent that from happening, as they'd have to deliberately first switch to global mode to change the default.

The "volume too loud for other users" problem I cited above. The reason why the video player was made local was an attempt at fixing "I don't want my ears blown out/I don't want to turn down the volume every time someone spawns a new one".

**Comment by TisFoolish** [comment]
> Or perhaps another option would be the following:
> 
> Default audio slider mode is local (purple), but the global volume isn't overridden until you touch it to make a change. Before it's touched, it's synced with the global volume. So if I want to set a default, I can switch to gloabal and just change the volume. Everyone else would see their (purple) volume slider move, until they touch it to change it, then they override it. This would actually fix a couple of problem:
> 
>     1. It would allow someone to deliberately change the default for everyone when spawning a new player
> 
>     2. Users who don't know how the slider works can't accidentally raise the volume for everyone, as by default they only set an override for themselves.
> 
>     3. Once you have an override set, it cannot be changed by others anymore, so a muted video can't be unmuted, and a unmuted video can't be muted by others.

That's what I was suggesting

**Comment by JackTheFoxOtter** [comment]
> The "volume too loud for other users" problem I cited above. The reason why the video player was made local was an attempt at fixing "I don't want my ears blown out/I don't want to turn down the volume every time someone spawns a new one".

But... this would basically fix that. It would only be 100% if the spawning user would first switch to global, and then change it to 100%. But by default, the volume could be something moderate like it already is currently. But someone walking up and changing volume by default wouldn't actually change it for everyone, only for themselves.

**Comment by BlueCyro** [comment]
> > The "volume too loud for other users" problem I cited above. The reason why the video player was made local was an attempt at fixing "I don't want my ears blown out/I don't want to turn down the volume every time someone spawns a new one".
> 
> But... this would basically fix that. It would only be 100% if the spawning user would first switch to global, and then change it to 100%. By default, the volume could be something moderate like it already is currently. But someone walking up and changing volume by default wouldn't actually change it for everyone, only for themselves.

The volume is currently not at 100% and this issue still is present, unfortunately.

**Comment by Frooxius** [comment]
> If the local slider is a multiplier, and the global audio is set to 0, then you wouldn't be able to actually raise your local volume though.

Yes, this is intentional. If you as host want to mute the sound or lower it for everyone (e.g. when someone talks), you can do that using this mechanism. The global mute could be just a simpler way of doing that too.

> I think the cloud variable to let the individual pick what volume state they want to subscribe to by default would probably be the more desirable option here.

I think this is very bad UX. It's like trying to fix convoluted problem by making it even more complicated.

---------------------------

I think it might be worth stepping back on this and seeing what are actually the issues and problems we want to solve there. There's a lot of back and forth here on things, but I get the feeling that everyone is talking about a bit of a different issue.

**Comment by BlueCyro** [comment]
> > If the local slider is a multiplier, and the global audio is set to 0, then you wouldn't be able to actually raise your local volume though.
> 
> Yes, this is intentional. If you as host want to mute the sound or lower it for everyone (e.g. when someone talks), you can do that using this mechanism. The global mute could be just a simpler way of doing that too.
> 
> > I think the cloud variable to let the individual pick what volume state they want to subscribe to by default would probably be the more desirable option here.
> 
> I think this is very bad UX. It's like trying to fix convoluted problem by making it even more complicated.
> 
> I think it might be worth stepping back on this and seeing what are actually the issues and problems we want to solve there. There's a lot of back and forth here on things, but I get the feeling that everyone is talking about a bit of a different issue.

I'll cite one of my previous comments for what I believe are the two opposing sides here:

- Local by default: "I don't want my ears blown out by other peoples' video players"

- Global by default: "I want to be able to turn my video players down for other people"

I personally very much disagree that the local volume should be a percentage of the global one for the issues with quieter audio setups and for those with hearing impairments.

**Comment by JackTheFoxOtter** [comment]
Yeah. My  underlying issue is the following:

When I spawn a video, currently I'm not able to mute it for everyone by default. I would like to be able to mute it for everyone by default.

> Yes, this is intentional. If you as host want to mute the sound or lower it for everyone (e.g. when someone talks), you can do that using this mechanism. The global mute could be just a simpler way of doing that too.

This wouldn't solve my problem, unfortunately. I wouldn't be able to by default set the volume to 0 unless someone opts in, since when I set the global volume to 0, noone can locally opt in. I do think the local setting to be an override makes the most sense.

**Comment by 5H4D0W-X** [comment]
I think a good simple compromise is to set the player to local, but have it start at 0 volume for every player. Then every player has to opt in to hearing the video, just as before when the spawning user set the global volume to 0, without having to set their volume slider to local first.

**Comment by 5H4D0W-X** [comment]
This is only a suggestion for personal players though, a video player without sound by default might be too confusing for new or inexperienced players

**Comment by JackTheFoxOtter** [comment]
> * I want to turn down video players down for other people - I think it'd be useful to understand this a bit more:
>   -- Does this mean only the initial state that the player starts in? Or even after people have already adjusted it? Can you elaborate in the scenarios where this occurs?

The common situation is that I spawn a video in a public world, and want to mute it for everyone in the world, unless they "opt in" to listen. It's usually when I want to show a friend a meme or some silly video that I don't want the other dozen people standing around to be forced to listen to. Right now, when anyone spawns a video, it will be on for everyone in the session, and can only be changed locally. That means everyone has to walk up to the video player and turn the volume down / off, which is highly disruptive.

>   -- Does "turn down" mean you want it to be quieter by default? Or just pure mute/un-mute?

Generally, I usually want to turn it off completely when showing a video. But I'm not sure a "mute for everyone" button would necessarily be the best option for everyone. That could cause other issues where I turned up the volume, then someone comes over and clicks the mute for everyone button because they don't want to listen to it, and now I have to configure it again. I think setting a default volume that ONLY applies when you haven't changed the volume slider manually yourself yet would be the most elegant solution.

**Comment by ShrikeAlvaron** [comment]
I think personally that having local volume be the default setting is for the best, it’s more intuitive generally and doesn’t have problems of new users not familiar with the toggles going and changing the volume without realizing they’re changing it for everyone.

it looks likes it’s been pointed out a few times, but I think the ideal to solve the issue of the video spawner wanting to set a considerate volume for other users would be to have the global volume slider change the default for the local volume, so the two would remain in sync for a user until they adjust the local volume and then have their own value, at which point we should respect their wishes of whatever volume they want it at anyway as that’s the point of having a local setting.

**Comment by JackTheFoxOtter** [comment]
> it looks likes it’s been pointed out a few times, but I think the ideal to solve the issue of the video spawner wanting to set a considerate volume for other users would be to have the global volume slider change the default for the local volume, so the two would remain in sync for a user until they adjust the local volume and then have their own value, at which point we should respect their wishes of whatever volume they want it at anyway as that’s the point of having a local setting.

Yes I agree. This is probably the most elegant solution you can have, still allowing someone to configure the default deliberately and intentionally, but removing the risk of users unintentionally affecting everyone.

**Comment by PointerOffset** [comment]
It's desirable for users to control how their content is presented (Global Volume). It's also desirable and a needed accessibility option to to allow users to tune the experience for their comfort and safety. We also don't want users to be confused about how the video player works. When you spawn in a video, one expects it to be clearly audible by default.

It's hard to meet all of these needs at the same time. But much of the problem comes from the fact that nobody can resolve all of these desires until it's too late. You need to hear the video to know if it's a problem.

**So what about adding a Volume Meter and spawning the player at a low volume? Something like 5%. The Volume Meter would show a clear visual indication of the absolute volume of a video while having it at a safely audible level by default.**

Considering this problem rarely gets mentioned for audio players, which have clear visual indications of volume, maybe it's worth considering if this would help?

**Comment by Frooxius** [comment]
There's additional issue though - if the volume is too low, then users might not even be aware of the player or audio and might not know they need to adjust it. 

There are scenarios where this is relevant - e.g. when the video is part of a presentation - having everyone turn it up so they can hear it is awkward. 

So collecting that, we have following problems:
1) Default global audio can be too loud for users
2) User might want to turn video down by default for everyone, so only people interested in it will be able to play
3) Users might need to ensure that video is loud enough by default for everyone to hear, without them having to turn it up to be able to hear and understand it in the first place (e.g. presentations)
4) Interaction needs to be simple for users who are just watching

Given these, it might be worth instead going away with the global/local volume toggle and have it local by default.

But then have UI element that will pop a separate slider to adjust the default volume for anyone who hasn't adjusted it yet as well as the default area mode. When the user adjusts these, that overrides it for them from that point on.

This default volume can be at average level (say 50 %) with local area. That on its own should address 1)

User can then turn the default volume down before playing (which addresses 2) ), or possibly up (or just leave it), which address 3)

When users want to make adjustments, they just fiddle with the slider, they don't need to worry about the defaults, which address 4)

**Comment by JackTheFoxOtter** [comment]
> Given these, it might be worth instead going away with the global/local volume toggle and have it local by default.

> But then have UI element that will pop a separate slider to adjust the default volume for anyone who hasn't adjusted it yet as well as the default area mode. When the user adjusts these, that overrides it for them from that point on.

This is basically the same thing I (and many others now) have suggested. Only difference being that I'd still use the local / global slider for this, as I think that's easier and more intuitive to operate. Purple: Your personal overrides, but synced with the default settings until you actually create an override by touching it. Blue: The default settings. You need to switch to them to deliberately configure them, and those then apply to everyone who hasn't yet created a personal override.

**Comment by PointerOffset** [comment]
I think that Froox's idea may simplify the interface for volume overall. I don't think anyone would be confused about a video having a default volume and it being adjusted to their needs only when adjusted. It sounds like a decent solution to me.

I might additionally suggest a known pattern I've seen in websites: Videos are muted by default, but have a clear indication they are muted and can be clicked on to un-mute them. It's a pattern I've seen in banner ads where they play a video but don't play audio unless you click the very clear 🔇 icon (or similar). Having such an icon overlaying the player when first spawned would create an opt-in scenario for hearing the video. That would allow an opportunity for someone to adjust the slider before opting-in to hear it. They could still see the video playing to get an idea if they might be in for something loud.

**Comment by Frooxius** [comment]
Yes, I'm just trying to collate the problems and possible solutions, to make sure that whatever is proposed hits all the underlying problems, rather than going back and forth on things.

The goal of having that default state being an explicit separate thing is to indicate that this is "Special/Distinct" configuration thing - it's not your volume, it's the default one. Almost like a "config" mode for the player if that makes sense.

**Comment by epicEaston197** [comment]
> Ah. Well this is not a good change in my eyes, as this makes it impossible to mute the video player for everyone currently in the world. This currently prevents me from spawning out any videos, as I don't want to blast anyone with audio they don't want to hear.

This was done for accessibility and safety reasons as some people when they hear a loud sound can lose their balance and potentially hurt themselves or maybe even go to the hospital firr has described it in their brain damage world

**Comment by JackTheFoxOtter** [comment]
> I might additionally suggest a known pattern I've seen in websites: Videos are muted by default, but have a clear indication they are muted and can be clicked on to un-mute them. It's a pattern I've seen in banner ads where they play a video but don't play audio unless you click the very clear 🔇 icon (or similar). Having such an icon overlaying the player when first spawned would create an opt-in scenario for hearing the video. That would allow an opportunity for someone to adjust the slider before opting-in to hear it. They could still see the video playing to get an idea if they might be in for something loud.

I don't think this applies very well to Resonite right now, because it's always a different player that brings in the video, not something that would appear on its own. If you think about the classroom type of situation again, you don't want the video to start muted if the lecturer pulls it out, you want it to be on by default for everyone in the class (world). This always depends on context, so I don't think there is a single good default that would work all of the time - that's why it's important to have the ability individually configure the default!

> The goal of having that default state being an explicit separate thing is to indicate that this is "Special/Distinct" configuration thing - it's not your volume, it's the default one. Almost like a "config" mode for the player if that makes sense.

Yes I understand that. For me, I always thought of the global slider as a "config mode" for the player already. Especially if the default mode is local, and you need to explicitly switch to a different mode (and color) to change the default configuration, so I felt this is already distinct enough, but as I said, I'm okay with either solution.

**Comment by JackTheFoxOtter** [comment]
> > Ah. Well this is not a good change in my eyes, as this makes it impossible to mute the video player for everyone currently in the world. This currently prevents me from spawning out any videos, as I don't want to blast anyone with audio they don't want to hear.
> 
> This was done for accessibility and safety reasons as some people when they hear a loud sound can lose their balance and potentially hurt themselves or maybe even go to the hospital firr has described it in their brain damage world

This is NOT a safety feature. For someone sensitive to loud sounds, they should ALWAYS change their client volume settings and not rely on in-world volume configurations. Also ironically this currently prevents me from turning off or lowering the audio for others currently if I want to prevent them being blased with it. I suggest reading the rest of the thread.

**Comment by JackTheFoxOtter** [comment]
> I would still like it to be local by default though my biggest pet peeve was people raising the default volume on the video player instead of making it local what you said would be a nice addition to that though

This is exactly what the proposed solution would ensure, I suggest reading the thread first.

**Comment by epicEaston197** [comment]
yeah i agree the video player should be local by default but until a user sets the volume it should be synced with the global volume slider
but this does have an issue that I brought up earlier of users just raising the global volume because they don't bother to click one button I don't wanna go up to a video player just to click on the local volume slider once and walk away just like my volume isn't raised by the global volume slider

**Comment by epicEaston197** [comment]
> Or perhaps another option would be the following:
> 
> Default audio slider mode is local (purple), but the global volume isn't overridden until you touch it to make a change. Before it's touched, it's synced with the global volume. So if I want to set a default, I can switch to gloabal and just change the volume. Everyone else would see their (purple) volume slider move, until they touch it to change it, then they override it. This would actually fix a couple of problem:
> 
>     1. It would allow someone to deliberately change the default for everyone when spawning a new player
> 
>     2. Users who don't know how the slider works can't accidentally raise the volume for everyone, as by default they only set an override for themselves.
> 
>     3. Once you have an override set, it cannot be changed by others anymore, so a muted video can't be unmuted, and a unmuted video can't be muted by others.

So I'm assuming that the user spawning out the video player will see it as local by default so the user would have to purposefully go out of their way to click the area mode button to raise the volume for everyone
I also hope that the volume is 0 by default

**Comment by JackTheFoxOtter** [comment]
> So I'm assuming that the user spawning out the video player will see it as local by default so the user would have to purposefully go out of their way to click the area mode button to raise the volume for everyone

Yes

> I also hope that the volume is 0 by default like others have said here

No (See above for reasoning behind this)

**Comment by epicEaston197** [comment]
> See above

please refer to a quote this thread is too long for me to remember what I've read

**Comment by JackTheFoxOtter** [comment]
> please refer to a quote this thread is too long for me to remember what I've read

References:
https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/2006#issuecomment-2099920769
> This would be fine for me if the default local volume was set to 0, so people have to "opt in" to listen. But I imagine that's not beneficial for other situations, like lectures where the lecturer spawns out a video and wants everyone to hear it by default.

https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/2006#issuecomment-2100087977
>> I might additionally suggest a known pattern I've seen in websites: Videos are muted by default, but have a clear indication they are muted and can be clicked on to un-mute them. It's a pattern I've seen in banner ads where they play a video but don't play audio unless you click the very clear 🔇 icon (or similar). Having such an icon overlaying the player when first spawned would create an opt-in scenario for hearing the video. That would allow an opportunity for someone to adjust the slider before opting-in to hear it. They could still see the video playing to get an idea if they might be in for something loud.
>
>I don't think this applies very well to Resonite right now, because it's always a different player that brings in the video, not something that would appear on its own. If you think about the classroom type of situation again, you don't want the video to start muted if the lecturer pulls it out, you want it to be on by default for everyone in the class (world). This always depends on context, so I don't think there is a single good default that would work all of the time - that's why it's important to have the ability individually configure the default!

**Comment by epicEaston197** [comment]
I want a solution that makes it so that when someone spawns on video they can't raise the default volume to Max

**Comment by epicEaston197** [comment]
I want a solution so that whenever someone spawns out a video they can't raise the default volume to Max because they don't bother to click one button

**Comment by JackTheFoxOtter** [comment]
> I want a solution so that whenever someone spawns out a video they can't raise the default volume to Max because they don't bother to click one button

With the proposed solution they'd have to go out of their way to change the default for everyone - which is a good thing. By default they'd just change their own local volume. But obviously if they WANT to change the default for everyone to max, they can. There's not much point in trying to obfuscate or lock that away though. There might be situations where that is desired, like a large lecture hall world.

**Comment by Veeridian** [comment]
So, I have some thoughts on the default video player's audio being set to local by default rather than global, after using it for a couple months.

At first, it would seem like having the volume set to local may cause less disruptions in sessions, however I'm finding the opposite is the case. An example from last night:

I had brought in a song that I had wished to show to a friend, and by default the local volume is set to 50%. Because it's set to local, and the local/global toggle is also local, this forced a few people to come from across the world to turn the local volume down. 

Prior, with the volume set to global by default, I could simply turn down the global volume to zero and set my local volume to whatever I'd want, disturbing no one, and only playing audio for interested audiences.

This is just a postulation of mine, but this may also result in new users confused on how to set the default video player to be global, say if they wished to use a looped or live-streamed video for music in their world. By default the video player's audio is set to be spatialized, and this setting is local too, so users may think they can't use the default video player to broadcast music. 

In short, I feel it benefits small groups of users who are all interested in watching a particular video, so they don't need to click the global/local toggle to set their local volume, but I see it being detrimental in most other use cases.

As far as any moderation-related concerns, there's little to stop someone from learning how to make their own loud disruption devices that don't rely on the default video player.

**Comment by ProbablePrime** [comment]
Can't we just make this a setting now and have the video player respond to that?

**Comment by epicEaston197** [comment]
> I had brought in a song that I had wished to show to a friend, and by default the local volume is set to 50%. Because it's set to local, and the local/global toggle is also local, this forced a few people to come from across the world to turn the local volume down.

I feel like this will be fixed just by lowering the volume to be about 25%

**Comment by epicEaston197** [comment]
> Can't we just make this a setting now and have the video player respond to that?

No that would be incorrect a setting like this would be a cloud variable that will be set via a facet that will be created by the content team similar to the seated mode option

Also see Issue #1800 This issue is bigger than the default media player I feel like if users had the choice to lower down music volume versus things like videos that would solve this issue somewhat

**Comment by ProbablePrime** [comment]
Easton, We could make this a setting setting. Instead of a cloud variable.

**Comment by epicEaston197** [comment]
> Easton, We could make this a setting setting. Instead of a cloud variable.

Yes I know but seated mode used to be a setting but was later removed because it was a cloud variable
I'm also wondering if this would be a generic implementation. will this just be an easily accessible value that we can reference elsewhere?.

For example we could have that slider control the local volume on a music player or whatever
